### DISCUSSION TITLE
Custom code

### DISCUSSION THREAD

**Discussion by dormasta** [joomengine/Joomla-Component-Builder#954] [Q&A] [answered]
Hello,
I am trying to reimport changes made in the IDE to JCB. Therefor I read and watched the tutorial custom code.
Those are the steps I do: 
1 - I added code in the PHP (controller methods) of the Custom Buttons
2- I compiled the component 
3- I replaced some code in the administration/components/com_component/controllers/view
4- I "marked" the code with 
/***[INSERT<>$$$$]***/		/***[/INSERT<>$$$$]***/ or
/***[REPLACE<>$$$$]***/		/***[/REPLACE<>$$$$]***/
5- recompiled
6- the code appears in custom code but following part doesn't appear in the IDE like in the tutorial. 
/***[INSERTED$$$$]***//*ID*/	 
 	<My CODE HERE>
/***[/INSERTED$$$$]***/
same with "replace"	 

What did I miss? any ideas? Thank you

**Comment by Llewellynvdm** [comment, marked as answer]
Well, for starters I will need more information to be able to say what is happening.

Some considerations:

- the placeholders are only added back if the component or compiler is set to do so:
   - in the compiling area
![image](https://user-images.githubusercontent.com/5607939/236978504-df9c7c26-54b3-4f0b-81b5-8f1987de35d7.png)
   - or in the component
![image](https://user-images.githubusercontent.com/5607939/236978574-c8a8da69-7a2e-4799-be96-679862179dc9.png)
- There are more than one tutorial on custom code, make sure you've watched them all
- We are working on the compiler so it could be a bug...

> This feature allows you to ship the component without the placeholders.

Should you think this is a bug, you must please open an issue here: https://git.vdm.dev/joomla/Component-Builder/issues

**Reply by dormasta** [reply]
Thank you, 
Indeed I missed to set those buttons, but I need to watch the other tutorials you mentioned as I don't get the expected result.
Cheers
